environment/config.go

```diff
@@ -51,6 +55,34 @@ type Configuration struct {
 
 	// StaticConfiguration is used for running M3DB with a static config
 	Static *StaticConfiguration `yaml:"static"`
+
+	// Presence of a (etcd) server in this config denotes an embedded cluster
+	EmbeddedServer *EmbeddedKV `yaml:"embeddedServer"`
+
+	// NamespaceTimeout is the timeout duration for setting a namespace
+	NamespaceTimeout time.Duration `yaml:"namespaceTimeout"`
+}
+
+// EmbeddedKV defines specific fields for the embedded kv server
+type EmbeddedKV struct {
+	Dir                string     `yaml:"dir"`
+	APUrls             []string   `yaml:"initialAdvertisePeerUrls"`
+	ACUrls             []string   `yaml:"advertiseClientUrls"`
+	LPUrls             []string   `yaml:"listenPeerUrls"`
+	LCUrls             []string   `yaml:"listenClientUrls"`
```

Comment:
Maybe we get rid of the acronymization of these variables?  Seems like they are inconsistent with the other long form variables below.